modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/security/IgniteSessionRepository.java

```diff
@@ -16,44 +16,61 @@
 
 package org.apache.ignite.console.web.security;
 
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
 import org.apache.ignite.Ignite;
-import org.apache.ignite.IgniteCache;
-import org.apache.ignite.cache.CacheMode;
-import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.console.db.CacheHolder;
+import org.apache.ignite.console.db.OneToManyIndex;
+import org.apache.ignite.console.dto.Account;
 import org.apache.ignite.console.messages.WebConsoleMessageSource;
+import org.apache.ignite.console.tx.TransactionManager;
 import org.springframework.context.support.MessageSourceAccessor;
 import org.springframework.session.ExpiringSession;
+import org.springframework.session.FindByIndexNameSessionRepository;
 import org.springframework.session.MapSession;
 import org.springframework.session.Session;
 import org.springframework.session.SessionRepository;
 
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+import static org.apache.ignite.console.common.Utils.getPrincipal;
 import static org.apache.ignite.console.errors.Errors.convertToDatabaseNotAvailableException;
 
 /**
  * A {@link SessionRepository} backed by a Apache Ignite and that uses a {@link MapSession}.
  */
-public class IgniteSessionRepository implements SessionRepository<ExpiringSession> {
-    /** */
-    private final Ignite ignite;
-
+public class IgniteSessionRepository implements FindByIndexNameSessionRepository<ExpiringSession> {
     /** Messages accessor. */
     private final MessageSourceAccessor messages = WebConsoleMessageSource.getAccessor();
 
     /** If non-null, this value is used to override {@link ExpiringSession#setMaxInactiveIntervalInSeconds(int)}. */
     private Integer dfltMaxInactiveInterval;
 
-    /** Session cache configuration. */
-    private final CacheConfiguration<String, MapSession> cfg;
+    /** */
+    private final TransactionManager txMgr;
+
+    /** Sessions collection. */
+    private CacheHolder<String, MapSession> sessionsCache;
+
+    /** Account to sessions index. */
+    private OneToManyIndex<String, String> accToSesIdx;
+
+    /** Sessions to account index. */
+    private OneToManyIndex<String, String> sesToAccIdx;
```

Comment:
Why you need this index? I see only usages in `save` and `delete`

Reply:
It is required for delete, because method `delete(sesId)` requires session ID.